spec/unit/webrtc/groupCall.spec.ts

```diff
@@ -0,0 +1,92 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { EventType, GroupCallIntent, GroupCallType, Room, RoomMember } from '../../../src';
+import { GroupCall } from "../../../src/webrtc/groupCall";
+import { MatrixClient } from "../../../src/client";
+import { MockAudioContext, MockMediaHandler } from '../../test-utils/webrtc';
+
+const FAKE_SELF_USER_ID = "@me:test.dummy";
+const FAKE_SELF_DEVICE_ID = "AAAAAA";
+const FAKE_SELF_SESSION_ID = "1";
+const FAKE_ROOM_ID = "!fake:test.dummy";
+
+describe('Group Call', function() {
+    beforeEach(function() {
+        // @ts-ignore Mock
+        global.AudioContext = MockAudioContext;
+    });
+
+    it("sends state event to room when creating", async () => {
+        const mockSendState = jest.fn();
+
+        const mockClient = {
+            sendStateEvent: mockSendState,
+            groupCallEventHandler: {
+                groupCalls: new Map(),
+            },
+        } as unknown as MatrixClient;
+
+        const room = new Room(FAKE_ROOM_ID, mockClient, FAKE_SELF_USER_ID);
+        const groupCall = new GroupCall(mockClient, room, GroupCallType.Video, false, GroupCallIntent.Prompt);
+
+        await groupCall.create();
+
+        expect(mockSendState.mock.calls[0][0]).toEqual(FAKE_ROOM_ID);
+        expect(mockSendState.mock.calls[0][1]).toEqual(EventType.GroupCallPrefix);
+        expect(mockSendState.mock.calls[0][2]["m.type"]).toEqual(GroupCallType.Video);
+        expect(mockSendState.mock.calls[0][2]["m.intent"]).toEqual(GroupCallIntent.Prompt);
+    });
+
+    it("sends member state event to room on enter", async () => {
+        const mockSendState = jest.fn();
+        const mockMediaHandler = new MockMediaHandler();
+
+        const mockClient = {
+            sendStateEvent: mockSendState,
+            groupCallEventHandler: {
+                groupCalls: new Map(),
+            },
+            callEventHandler: {
+                calls: new Map(),
+            },
+            mediaHandler: mockMediaHandler,
+            getMediaHandler: () => mockMediaHandler,
+            getUserId: () => FAKE_SELF_USER_ID,
+            getDeviceId: () => FAKE_SELF_DEVICE_ID,
+            getSessionId: () => FAKE_SELF_SESSION_ID,
+            emit: jest.fn(),
+            on: jest.fn(),
+            removeListener: jest.fn(),
+        } as unknown as MatrixClient;
+
+        const room = new Room(FAKE_ROOM_ID, mockClient, FAKE_SELF_USER_ID);
+        const groupCall = new GroupCall(mockClient, room, GroupCallType.Video, false, GroupCallIntent.Prompt);
+
+        room.currentState.members[FAKE_SELF_USER_ID] = {
+            userId: FAKE_SELF_USER_ID,
+        }as unknown as RoomMember;
+
+        await groupCall.create();
+
+        await groupCall.enter();
+
+        expect(mockSendState.mock.lastCall[0]).toEqual(FAKE_ROOM_ID);
+        expect(mockSendState.mock.lastCall[1]).toEqual(EventType.GroupCallMemberPrefix);
```

Comment:
We should check that the state event contents match up with what we're expecting, too